cpp/open3d/visualization/rendering/filament/FilamentRenderToBuffer.cpp

```diff
@@ -133,9 +137,11 @@ void FilamentRenderToBuffer::SetDimensions(const std::uint32_t width,
     width_ = width;
     height_ = height;
 
-    buffer_size_ = width * height * 3 * sizeof(std::uint8_t);
+    buffer_size_ = width * height * n_channels_ * sizeof(std::uint8_t);
     if (buffer_) {
         buffer_ = static_cast<std::uint8_t*>(realloc(buffer_, buffer_size_));
+    } else {
+        buffer_ = static_cast<std::uint8_t*>(malloc(buffer_size_));
```

Comment:
shouldn't be uint16_t? If the depth buffer are encoding float numbers...

Reply:
No, we're reading from the color buffer, which is a uint8_t*. The shader encodes the (most significant 24-bits of the) float into the R, G, and B channels. As far as OpenGL and Filament are concerned, this is an ordinary RGBA32 color buffer.